planner/core/expression_rewriter.go

```diff
@@ -562,7 +562,7 @@ func (er *expressionRewriter) handleExistSubquery(v *ast.ExistsSubqueryExpr) (as
 			er.err = errors.Trace(err)
 			return v, true
 		}
-		if len(rows) > 0 {
+		if (len(rows) > 0 && !v.Not) || (len(rows) == 0 && v.Not) {
```

Comment:
what does this check mean?

Reply:
`rows` is the result of the subquery evaluated by `EvalSubquery()`, it directly set the result of `exists` subquery according to `rows`.